modules/imap/functions.php

```diff
@@ -1144,3 +1145,17 @@ function get_personal_ns($imap) {
     );
 }}
 
+/**
+ * @subpackage imap/functions
+ */
+if (!hm_exists('check_message_part')) {
+function check_message_part($part) {
```

Comment:
Please fix this issue in hm-imap-bodystructure file where this array is generated instead of swapping parameters later. It feels odd to do this here.

We probably have a bug in set_value or attribute_set methods in that Hm_IMAP_Struct class but we need to make sure it returns proper structure with fields that are not swapped.

Reply:
<img width="782" alt="Screenshot 2023-07-06 at 22 04 15" src="https://github.com/cypht-org/cypht/assets/80334370/efa4d56d-5964-49ee-b164-03fe6291429f">

After debugging I realized that the bug comes from the message itself. Please check.